config/structs.go

```diff
@@ -36,6 +36,13 @@ type ListenConfig struct {
 	Port            int
 	Address         string
 	MetricsEndpoint string `hcl:"metrics_endpoint" yaml:"metrics_endpoint"`
+	TLS				TLSConfig
```

Comment:
Also, this feature should be documented somewhere. Maybe just add it to the example config in the README?